src/vision/status_quo/alan_started_trusting_the_rust_compiler.md

```diff
@@ -0,0 +1,59 @@
+# 😱 Status quo stories: Alan started trusting the Rust compiler
+
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+## The story
+### Trust the compiler
+Alan has a a lot of experience in _GC'd language_, but in the meantime has created some successful projects in Rust.
+He has dealt with his fair share of race conditions/thread safety issues during runtime in _GC'd language_, but is now starting to trust that if his Rust code compiles,
+he won't have those annoying runtime problems to deal with.
+
+This allows him to try to squeeze his programs for as much performance as he wants, because the compiler will stop him when he tries things that could result in runtime problems.
+After seeing the perfomance and the lack of runtime problems, he starts to trust the compiler more and more with each project finished.
+
+He knows what he can do with external libraries, he does not need to fear concurrency issues if the library cannot be used from multiple threads, because the compiler would tell him.
+
+His trust in the compiler solidifies further the more he codes in Rust.
+
+### The first async project
+Alan now starts with his first async project. He sees that there is no async in the standard library, but finds a crate that provides some async versions of the standard library functions.
+He has some code written, but the compiler complains that `await` is only allowed in `async` functions. He now notices that all the examples use `#[async-version-of-std::main]` 
+as an attribute on the `main` function in order to be able to turn it into an `async main`, so he does the same to get his code compiling.
+
+This aligns with what he knows from _GC'd language_, where you also change the entry point of the program to be async, in order to use `await`.
+Everything is great now, the compiler is happy, so no runtime problems, so Alan is happy.
+
+The project is working like a charm.
+
+### Fractured futures, fractured trust
+The project Alan is building is starting to grow, and he decides to add a new feature. He starts using _async-library-that-does-not-run-on-current-executor_ in order to help him achieve this task.
+After a lot of refactoring to make the compiler accept the program again, Alan is satisfied that his refactoring is done.
+He runs his project but is suddenly greeted with a runtime error? How is this even possible? His project doesn't contain any out-of-bounds accesses, he never uses `.unwrap` or `.expect`?
+At the top of the error message he sees: `This async thing needs to run in the context of this other async library, and not the one you are using.` 
```

Comment:
Can you show the actual output? Like with backticks?

Reply:
Here's probably the most straight-forward way that a new user might stumble:

First, they copy in the example from the [async-std README](https://crates.io/crates/async-std).  This works just fine.  Next, they modify it slightly by copying in some example code from the [hyper client docs](https://hyper.rs/guides/client/basic/).   Note that the hyper docs do say that tokio is needed (thus Alan will have also added tokio to Cargo.toml), but it's not clear that tokio and async-std will conflict.

The resulting code seems innocent enough:

```rust
use async_std::task;
use hyper::Client;

async fn say_hello() {
    println!("Hello, world!");

    let client = Client::new();
    let uri = "http://httpbin.org/ip".parse().unwrap();
    let resp = client.get(uri).await.unwrap();
    println!("Response: {}", resp.status());

}

fn main() {
    task::block_on(say_hello())
}
```

But it will fail at runtime:

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.11s
     Running `target\debug\foo.exe`
Hello, world!
thread 'main' panicked at 'there is no reactor running, must be called from the context of a Tokio 1.x runtime', C:\Users\achin\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.4.0\src\runtime\blocking\pool.rs:85:33
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: process didn't exit successfully: `target\debug\foo.exe` (exit code: 101)
```

Reply:
This is perfect!

Reply:
@eminence, this is not too far from the problems that I originally had, although your example is a lot cleaner/more concise than what I originally had, do you mind if I take it over and slightly modify it for in the PR?

Reply:
I don't mind at all, please include and much or as little of my comment in your PR as you want